src/Confirmation.vala

```diff
@@ -19,18 +19,51 @@
 */
 
 public class Notifications.Confirmation : AbstractBubble {
+    private string _confirmation_type;
+    public string confirmation_type {
+        get {
+            return _confirmation_type;
+        }
+        set {
+            if (value == _confirmation_type) {
+                return;
+            }
+
+            _confirmation_type = value;
+
+            if (icon_name_binding != null) {
+                icon_name_binding.unbind ();
+            }
+            if (progress_binding != null) {
+                progress_binding.unbind ();
+            }
+
+            var contents = create_contents ();
+            content_area.add (contents);
+            content_area.visible_child = contents;
+        }
+    }
+
     public new string icon_name { get; construct set; }
     public double progress { get; construct set; }
 
-    public Confirmation (string icon_name, double progress) {
+    private Binding? icon_name_binding;
+    private Binding? progress_binding;
+
+    public Confirmation (string confirmation_type, string icon_name, double progress) {
         Object (
+            confirmation_type: confirmation_type,
             icon_name: icon_name,
             progress: progress,
             timeout: 2000
         );
     }
 
     construct {
+        get_style_context ().add_class ("confirmation");
```

Comment:
this can be moved to be done after the `Object()` call, and the construct block removed.